core/src/main/java/com/linecorp/armeria/common/logging/RequestContextExporterBuilder.java

```diff
@@ -15,34 +15,23 @@
  */
 package com.linecorp.armeria.common.logging;
 
+import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
-import java.util.HashSet;
-import java.util.Set;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.function.Function;
+import java.util.stream.Collectors;
 
-import javax.annotation.Nullable;
-
-import com.linecorp.armeria.common.HttpHeaderNames;
-import com.linecorp.armeria.common.logging.RequestContextExporter.ExportEntry;
-
-import io.netty.util.AsciiString;
 import io.netty.util.AttributeKey;
 
 /**
  * Builds a new {@link RequestContextExporter}.
  */
 public final class RequestContextExporterBuilder {
 
-    static final String PREFIX_ATTRS = "attrs.";
-    private static final String ATTR_NAMESPACE = "attr:";
-    private static final String PREFIX_REQ_HEADERS = "req.headers.";
-    private static final String PREFIX_RES_HEADERS = "res.headers.";
-
-    private final Set<ExportEntry<BuiltInProperty>> builtIns = new HashSet<>();
-    private final Set<ExportEntry<AttributeKey<?>>> attrs = new HashSet<>();
-    private final Set<ExportEntry<AsciiString>> reqHeaders = new HashSet<>();
-    private final Set<ExportEntry<AsciiString>> resHeaders = new HashSet<>();
+    private final ExportGroupBuilder defaultExportGroupBuilder = ExportGroup.builder();
+    private final List<ExportGroup> exportGroups = new ArrayList<>();
```

Comment:
I think `<exportGroup>` is an optional feature. Don't we need to initialize this by default.
How about make it `@Nullable` and initailzed it when it is actually used.